openvasd/openvasd.c

```diff
@@ -0,0 +1,2323 @@
+};
+
+/**
+ * @brief Struct holding vt information
+ */
+struct openvasd_vt_single
+{
+  gchar *vt_id;
+  GHashTable *vt_values;
+};
+
+/**
+ * @brief Request methods
+ */
+enum openvas_request_method
+{
+  POST,
+  GET,
+  HEAD,
+  DELETE,
+};
+
+typedef enum openvas_request_method openvasd_req_method_t;
+
+/** @brief Initialize an openvasd connector.
+ *
+ *  @return An an openvasd connector struct.
+ */
+openvasd_connector_t
+openvasd_connector_new (void)
+{
+  openvasd_connector_t connector;
+  connector = g_malloc0 (sizeof (struct openvasd_connector));
+  return connector;
+}
+
+/** @brief Build a openvasd connector
+ *
+ *  Receive option name and value to build the openvasd connector
+ *
+ *  @param conn struct holding the openvasd connector information
+ *  @param opt    option to set
+ *  @param val    value to set
+ *
+ *  @return Return OK on success, otherwise error;
+ */
+openvasd_error_t
+openvasd_connector_builder (openvasd_connector_t *conn, openvasd_conn_opt_t opt,
+                            const void *val)
+{
+  if (*conn == NULL)
+    *conn = openvasd_connector_new ();
+
+  if (opt < OPENVASD_CA_CERT || opt > OPENVASD_PORT)
+    return OPENVASD_INVALID_OPT;
+
+  if (val == NULL)
+    return OPENVASD_INVALID_VALUE;
+
+  switch (opt)
+    {
+    case OPENVASD_CA_CERT:
+      (*conn)->ca_cert = g_strdup ((char *) val);
+      break;
+    case OPENVASD_CERT:
+      (*conn)->cert = g_strdup ((char *) val);
+      break;
+    case OPENVASD_KEY:
+      (*conn)->key = g_strdup ((char *) val);
+      break;
+    case OPENVASD_API_KEY:
+      (*conn)->apikey = g_strdup ((char *) val);
+      break;
+    case OPENVASD_SERVER:
+      (*conn)->server = g_strdup ((char *) val);
+      break;
+    case OPENVASD_HOST:
+      (*conn)->host = g_strdup ((char *) val);
+      break;
+    case OPENVASD_SCAN_ID:
+      (*conn)->scan_id = g_strdup ((const gchar *) val);
+      break;
+    case OPENVASD_PORT:
+    default:
+      (*conn)->port = *((int *) val);
+      break;
+    };
+
+  return OPENVASD_OK;
+}
+
+/** @brief Build a openvasd connector
+ *
+ *  Receive option name and value to build the openvasd connector
+ *
+ *  @param conn   struct holding the openvasd connector information
+ *
+ *  @return Return OPENVASD_OK
+ */
+openvasd_error_t
+openvasd_connector_free (openvasd_connector_t *conn)
+{
+  if (*conn == NULL)
+    return OPENVASD_OK;
+
+  g_free ((*conn)->ca_cert);
+  g_free ((*conn)->cert);
+  g_free ((*conn)->key);
+  g_free ((*conn)->apikey);
+  g_free ((*conn)->server);
+  g_free ((*conn)->host);
+  g_free ((*conn)->scan_id);
+  g_free (*conn);
+  *conn = NULL;
+
+  return OPENVASD_OK;
+}
+
+/**
+ * @brief Free an openvasd response struct
+ *
+ * @param resp Response to be freed
+ */
+void
+openvasd_response_cleanup (openvasd_resp_t resp)
+{
+  if (resp == NULL)
+    return;
+
+  g_free (resp->body);
+  g_free (resp->header);
+  g_free (resp);
+  resp = NULL;
```

Comment:
Another one of these redundant assignments. `resp` is a pointer, so this is like assigning to an int argument.